### DISCUSSION TITLE
Kafka exporter logging level

### DISCUSSION THREAD

**Discussion by jesussancheztellomm** [strimzi/strimzi-kafka-operator#5517] [Q&A] [answered]
Hi,

Is there a way to reduce the logging level of the Kafka Exporter? I am using Strimzi 0.24.1

I can see a KAFKA_EXPORTER_LOGGING env variable in the container, but i haven't find a way to set it to ERROR in the Kafka spec.

Thanks in advance,
All the best
Jesús

**Comment by scholzj** [comment, marked as answer]
You should be able to configure it in the `.spec.kafkaExporter` section: https://strimzi.io/docs/operators/latest/full/using.html#type-KafkaExporterSpec-reference

**Comment by jesussancheztellomm** [comment]
Thanks! Looks like there is some kind of bug because i have set it to error but still see info messages in the log.

    Environment:
      KAFKA_EXPORTER_LOGGING:        error
      KAFKA_EXPORTER_KAFKA_VERSION:  2.7.0
      KAFKA_EXPORTER_GROUP_REGEX:    .*
      KAFKA_EXPORTER_TOPIC_REGEX:    .*
      KAFKA_EXPORTER_KAFKA_SERVER:   cadence-master-kafka-bootstrap:9091
      KAFKA_EXPORTER_ENABLE_SARAMA:  false

  [kafka_exporter] [INFO] 2021/09/03 06:23:04 Starting kafka_exporter (version=, branch=, revision=)
  [kafka_exporter] [INFO] 2021/09/03 06:23:04 Build context (go=go1.16.5, user=, date=)
  [kafka_exporter] [INFO] 2021/09/03 06:23:04 Done Init Clients
  [kafka_exporter] [INFO] 2021/09/03 06:23:04 Listening on :9404
  [kafka_exporter] [INFO] 2021/09/03 06:23:20 Refreshing client metadata
  [kafka_exporter] [INFO] 2021/09/03 06:23:20 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:22 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:28 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:30 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:38 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:40 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:42 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:48 Fetching consumer group metrics
  [kafka_exporter] [INFO] 2021/09/03 06:23:50 Fetching consumer group metrics

**Reply by scholzj** [reply]
Hmm, maybe it is some bug ... I know we had some issues with logging in the fixed Kafka Exporter introduced in 0.24+ ... I will try to remember if we fixed this at the end or not.

**Reply by scholzj** [reply]
Looks like you are right. I opened #5529 for this.

**Reply by sknot-rh** [reply]
Just a clarification. Go logging works differently from Java. `Error`, `Warn` and `Fatal` log messages are printed regardless of the configured logging level. We can just configure verbosity of the `info` level. In https://github.com/strimzi/strimzi-kafka-operator/pull/5575 I updated KafkaExporter to version, which supports this granularity. However, the supported log levels are just `info`, `debug` and `trace`.
